icon_validator/rules/plugin_validators/help_input_output_validator.py

```diff
@@ -4,6 +4,91 @@
 from icon_validator.rules.validator import KomandPluginValidator
 from icon_validator.exceptions import ValidationException
 
+from datetime import datetime
+
+
+def detect_valid_datetime(list_entry: str) -> bool:
+    """
+    This function determines if a valid datetime following
+    iso format is detected in a string
+
+    :param list_entry: The entry of the list to be checked for datetime validity
+    :type list_entry: str
+
+    :rtype: bool
+    """
+    try:
+        datetime.fromisoformat(list_entry)
+        return True
+    except ValueError:
+        return False
+
+
+def convert_to_valid_datetime(date: str) -> str:
+    """
+    This function takes a str containing a date in isoformat and converts
+    it to the format that matches the format found within the raw help md
+
+    :param date: datetime string
+    :type date: str
+
+    :return final_result: datetime string containing 'T'
+    :rtype final_result: str
+    """
+    # Convert the string to a datetime object so we can work with it
+    datetime_object = datetime.fromisoformat(date)
+
+    # Create the datetime containing the T using strftime
+    new_datetime = datetime_object.strftime("%Y-%m-%dT%H:%M:%S")
+
+    # Extract the timezone separately
+    timezone = datetime_object.strftime("%z")
+
+    # If timezone was not found in the string, then just produce the new amended datetime
+    if timezone == '':
+
+        final_result = new_datetime
+
+    else:
+        # Add the ':' in manually because strftime does not contain an option to convert it like this
+        new_timezone = timezone[:3] + ":" + timezone[3:]
+
+        # Combine new datetime and append the amended timezone on the end
+        final_result = new_datetime + new_timezone
+
+    # Return the final result
+    return final_result
+
+
+def datetime_checker_fixer(action_input: str):
+    """
+    This function uses the other two to perform the actions needed to detect and
+    convert the datetime to correct format
+
+    :param action_input: table string for the action from help md
+    :type action_input: str
+    """
+
+    # Take the table string, and split it by the '|' character
+    split_action_input = action_input.split("|")
+
+    # Loop through each entry in the newly created list from split
+    for entry in split_action_input:
+
+        # Use detect function to detect which entry contains the date
+        if detect_valid_datetime(entry):
+
+            # When found, use convert function to convert the list element to the format we need
+            new_datetime_value = convert_to_valid_datetime(entry)
+
+            # Replace the old value with the new one
+            split_action_input[-2] = new_datetime_value
```

Comment:
Is it always `-2`? 😃

Reply:
In the case of the example field, I believe the example always comes last. In this case it is `[-2]` because the last entry when the table string is split is `''`. 
Do you know of any examples where example isn't the last field in the string?

Reply:
I think I don't know. That's fine then! 😃